### PR TITLE
move to I/O Gateway, validate Gateway service token

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
- validation of GW token
- changes in authorization header (use basic auth header with base64 encoding, like OpenWhsik)
- remove versioning
- doc changes
- remove obsolete functional tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/Tvm.js [hunk 1]
```diff
@@ -154,11 +237,23 @@ class Tvm {
    */
   async processRequest (params) {
     try {
+      // 0. validate Adobe I/O Api Gateway service token
+      if (params.disableAdobeIOApiGwTokenValidation !== 'true') {
```
Comment (sandeep-paliwal) [inline lib/Tvm.js:241]: Can we have this flag passed as boolean rather than string?

Comment (moritzraho) [reply lib/Tvm.js:241]: Not sure if can do this in the manifest default params, for now everything comes in as a string

Comment (moritzraho) [reply lib/Tvm.js:241]: actually in the wskdeploy specs there is a type parameter https://github.com/apache/openwhisk-wskdeploy/blob/master/specification/html/spec_parameters.md#fields, checking if we support this in our tooling..

Comment (moritzraho) [reply lib/Tvm.js:241]: so I did some testing and params type are correctly parsed and set in `params` in those cases:
```yaml
inputs:
  a: 123 # integer
  b: true # boolean
  c: 'abc' # string
```

However the problem is that `disableAdobeIOApiGwTokenValidation` is set through an env variable `DISABLE_ADOBE_IO_API_GW_TOKEN_VALIDATION` and that environment variables always evaluate to strings.

And I also tried the following yaml syntax
```yaml
inputs:
  a: 
    type: integer
    value: $A_ENV # A_ENV=123
```
The deployed action gets `params.a='$A_ENV'`

So for now  we will have to continue with strings I guess

**Pull request merged**